Changes examples to use new node builder interface.

eelcocramer opened the pull request:
Fixes #28

eelcocramer committed:
Changes examples to use new node builder interface.

jbenet commented:
@eelcocramer thanks

please use gofmt -- your PR has a bunch of changes that are just changing whitespace from tabs to spaces, something that gofmt is against.

eelcocramer committed:
Fixes fmt using gofmt

eelcocramer commented:
I already thought that might be a problem :-p I'm new to go and didn't know about gofmt. I've fixed the files and added the commit to the PR.

Thanks.

whyrusleeping commented on examples/api/demoapp/demo.go:
```diff
@@ -26,13 +26,16 @@ func CountChars(r io.Reader) map[byte]int {
 
-	builder := core.NewNodeBuilder().Online().SetRepo(r)
-	return builder.Build(context.Background())
+	cfg := new(core.BuildCfg)
```
I would prefer to use the composite construction here:

``` go
cfg := &core.BuildCfg{
    Repo: r,
    Online: true,
}
```

eelcocramer committed:
Improves cfg creation and fixes readme's

eelcocramer commented:
That is more elegant.

jbenet commented on examples/api/demoapp/demo.go:
```diff
@@ -1,15 +1,15 @@
 
 import (
-	"bufio"
+    "bufio"
```
`gofmt`. please run `gofmt` every time. add it to your editor.

eelcocramer replied:
Thought I had that figured out. Apparently not. Will add a commit momentarily. Thanks.

jbenet commented on examples/api/demoapp/readme.md:
```diff
@@ -10,15 +10,15 @@ To Start off, lets get some imports:
+    "github.com/ipfs/go-ipfs/core"
+    "github.com/ipfs/go-ipfs/core/coreunix"
+    "github.com/ipfs/go-ipfs/repo/fsrepo"
+    "code.google.com/p/go.net/context"
```
go fmt.

eelcocramer replied:
So, gofmt does nothing on the readme.

eelcocramer committed:
Gofmts demo.go

eelcocramer committed:
Gofmts service example

eelcocramer commented:
Ok, format is fixed again. Seems that I imported a vim bundle from the wrong location. I'm ready to "go" now... :-)

eelcocramer committed:
Improves readabilty for readme's

eelcocramer committed:
Gofmt's example code in readme's

jbenet commented on examples/api/readme.md:
```diff
@@ -63,21 +41,31 @@ import "github.com/ipfs/go-ipfs/repo/fsrepo"
+### Context
+If youve never dealt with contexts before, I highly recommend you first go read
+[this wonderful explanation](https://blog.golang.org/context). Now, the context
+we pass into `Build` is the "master" context to the entire ipfs node, cancelling
```
`Build` is no longer a thing, right?

eelcocramer replied:
Missed that one... Thanks.

jbenet commented:
> I'm ready to "go" now... :-)

hahaha :) 

LGTM above, except the one stale ref to `Build`

eelcocramer committed:
Removes stale reference to 'Build'

jbenet commented:
thanks @eelcocramer

eelcocramer commented:
Thank you @jbenet for creating ipsf in the first place!
